src/js/flows/workspace/initWorkspace.ts

```diff
@@ -0,0 +1,18 @@
+import Current from "../../state/Current"
+import Workspaces from "../../state/Workspaces"
+import {globalDispatch} from "../../state/GlobalContext"
+import {Workspace} from "../../state/Workspaces/types"
+import WorkspaceStatuses from "../../state/WorkspaceStatuses"
+import {WorkspaceStatus} from "../../state/WorkspaceStatuses/types"
+import refreshSpaceNames from "../refreshSpaceNames"
+
+export const initWorkspace = (ws: Workspace, status: WorkspaceStatus) => (
+  dispatch
+): void => {
+  dispatch(Workspaces.add(ws))
+  dispatch(WorkspaceStatuses.set(ws.id, status))
+  globalDispatch(Workspaces.add(ws)).then(() => {
+    dispatch(Current.setWorkspaceId(ws.id))
+    dispatch(refreshSpaceNames())
+  })
```

Comment:
`initWorkspace` commits the given workspace to state, updates its status, and sets it to current so the user is taken to it. It is called by the workspace form to create/update a workspace after the form has already pinged, assembled, and potentially authenticated the provided workspace.

`activateWorkspace` is always used for an existing workspace that we want to switch to, but may be stale in some ways (such as version/availability, or it's accessToken has not been loaded into state from the secret storage yet), in which case we refresh that data if possible. We call this whenever we are loading a workspace we've created before (The workspace chooser menu, the page version of that chooser, login page, and loading a tab that already has a workspace open)

Calling `initWorkspace` instead of `activateWorkspace` would result in stale workspace data, and calling `activateWorkspace` instead of `initWorkspace` would make redundant/excessive api calls (since the form will have already done that part).

Reply:
Thanks for the explanation Mason. That makes sense. I think the words init and activate are too related. Maybe we have a `saveWorkspace` does everything that "activateWorkspace" doesn't.

If so, then when a user first creates a workspace the flow is: 1) save workspace 2) activate workspace

And when they click on an existing workspace: 1) activate workspace

I'd argue that it's better to have 2 or 3 redundant API calls if the code is more strait forward.